middlewares/auth.middleware.js

```diff
@@ -249,6 +272,66 @@ function deleteResetToken(req, res, next) {
     );
 }
 
+/**
+ * Attempts to find a rolebinding given the role name and adds it to the account
+ * @param {ObjectId} accountId the id of the account that you want to add a rolebinding to
+ * @param {String} roleName the name of the role that you want to add
+ */
+async function createRoleBindingByRoleName(accountId, roleName){
+    const role = await Services.Role.getRole(roleName);
+    if (!!role) {
+        await Services.RoleBinding.createRoleBinding(accountId, role.id);
+    }
+}
+
+/**
+ * Middleware that creates rolebinding to access POST route for respective account
+ * @param {{body: {account:{accountType:String, id: ObjectId}}}} req the request object
+ * @param {*} res 
+ * @param {(err?)=>void} next 
+ */
+async function addCreationRoleBindings(req, res, next){
+    // Get the default role for the account type given
+    const roleName = Constants.POST_ROLES[req.body.account.accountType];
```

Comment:
@YiFeiZhang2 is this a valid assumption that the accountType is a `Role`?

Reply:
Also, when is this used?

Reply:
All the accountTypes had role name which corresponded
this is used to add roles to let the accounts POST to the respective subtype